apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java

```diff
@@ -324,14 +324,7 @@ public AbstractSpan createExitSpan(final String operationName, final String remo
             exitSpan = parentSpan;
         } else {
             final int parentSpanId = parentSpan == null ? -1 : parentSpan.getSpanId();
-            if (StringUtil.isEmpty(remotePeer)) {
```

Comment:
This deletion actually is not related to this PR. The lazy peer initialization is another feature :)